src/unleashclient.cpp

```diff
@@ -61,12 +68,19 @@ void UnleashClient::initializeClient() {
         // Initial fetch of feature flags
         auto apiFeatures = m_apiClient->features();
         if (apiFeatures.empty()) {
-            std::cerr << "Attempted to initialize an Unleash Client instance "
-                         "without server response."
-                      << std::endl;
-            return;
+            std::cerr << "Attempted to initialize an Unleash Client instance without server response." << std::endl;
+            std::ifstream cacheFile(m_cacheFilePath, std::fstream::in);
+            if (cacheFile.is_open()){
+                std::cout << "Reading configuration from cached file " << m_cacheFilePath << std::endl;
+                std::stringstream features_buffer;
+                features_buffer << cacheFile.rdbuf();
+                cacheFile.close();
+                m_features = loadFeatures(features_buffer.str());
+            } else 
+                std::cout << "Could not open cache file '" << m_cacheFilePath << "' for reading." << std::endl;
```

Comment:
There is a missing `return` in this case. Otherwise the periodic task is launched and the instance is marked as initialized.